test/model/test_store.dart

```diff
@@ -76,6 +77,16 @@ class TestGlobalStore extends GlobalStore {
 
   @override
   Future<Account> doInsertAccount(AccountsCompanion data) async {
+    // Check for duplication is typically handled by the database but since
+    // we're not using a real database, this needs to be handled here.
+    // See [AppDatabase.createAccount].
```

Comment:
There's now a little bit of asymmetry in that the `TestGlobalStore` throws `AccountAlreadyExistsException` whereas its thrown by the database in real usage, so I added a comment to call that out.